include/service_config.h

```diff
@@ -28,10 +28,15 @@ struct genotyper_config {
     /// Output format (default = bcf), choices = "BCF", "VCF"
     GLnexusOutputFormat output_format = GLnexusOutputFormat::BCF;
 
+    // Should the genotyper write a record describing each call loss?
+    // If true, the output is recorded in YAML format in the
+    // [residuals_file].
+    bool output_residuals = false;
+    std::string residuals_file = "/tmp/residuals.yml";
```

Comment:
Let's just add ".residuals.yml" to the output BCF filename in Service::genotype_sites, instead of taking an explicit file path.